sql/src/main/java/io/crate/operation/collect/LuceneShardCollectorProvider.java

```diff
@@ -121,28 +121,25 @@ public OrderedDocCollector getOrderedCollector(RoutedCollectPhase phase,
                                                    boolean requiresRepeat) {
         RoutedCollectPhase collectPhase = phase.normalize(shardNormalizer, null);
 
-        CrateSearchContext searchContext = null;
         CollectorContext collectorContext;
         CollectInputSymbolVisitor.Context ctx;
+        Engine.Searcher searcher = null;
+        LuceneQueryBuilder.Context queryContext;
         try {
-            searchContext = searchContextFactory.createContext(
-                sharedShardContext.readerId(),
-                sharedShardContext.indexShard(),
-                sharedShardContext.searcher(),
-                collectPhase.whereClause()
+            searcher = sharedShardContext.acquireSearcher();
```

Comment:
that would also close it in the success case. But in that case it must be kept open - it is added to the JobCollectContext which takes care of closing it.